### PR TITLE
created category pages and filter/sort products

### PR DESCRIPTION
**Pull request by Ksenaaa**

### CODE REVIEW

File: routes/product.routes.js [hunk 1]
```diff
@@ -1,24 +1,19 @@
...
+const paginationByElement = require('../utils/paginationByElement')
+const paginationProductsList = require('../utils/paginationProductsList')
+const productsCursor = require('../utils/productsCursor')
 
 router.get(
   '/new',
   async (req, res) => {
     try {
-      const { dataModel } = await lastItem(productModel, 6)
+      const { dataModel } = await lastItems(productModel, 6)
```
Comment (vell28) [inline routes/product.routes.js]: 6 - magic variable

File: utils/createFilterData.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  const productPrice = productsModel.map(element => element.quality.standart)
+  const productWidth = productsModel.map(element => element.characteristics.size.width)
+  const productHeight = productsModel.map(element => element.characteristics.size.height)
+  const productLength = productsModel.map(element => element.characteristics.size.length)
+
+  const createFilterElementCheckbox = (data, nameElement) => {
+    let resultValue = {}
+
+    data.sort().forEach(element => {
+      resultValue[element] = resultValue[element] + 1 || 1
```
Comment (vell28) [inline utils/createFilterData.js]: strange logic, take a look again pls

File: utils/createFilterData.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      valueElement = filterParse[nameElement].reduce((acc, item) => {
+        acc[item.split(':')[0]] = Number(item.split(':')[1])
+        return acc
+      }, {})
+
+      const filterElement = productValue.filter(element => {
+        const from = valueElement.from
+        const to = valueElement.to
+
+        return from && to ? element >= from && element <= to :
```
Comment (vell28) [inline utils/createFilterData.js]: hard to read logic, please split it

File: utils/sortProducts.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+const sortProducts = (sortData) => {
+  if (sortData) {
+    let parseSort = {}
+
+    sortData.split(';').forEach(element => {
+      const newElement = element.split('=')
+      return parseSort[newElement[0]] = newElement[1]
+    })
+    
+    const sortField = parseSort.option === 'price' ? 'quality.standart' :
```
Comment (vell28) [inline utils/sortProducts.js]: hard to read logic, split it

File: utils/createFilterData.js (hunk 3 above)
Comment (Ksenaaa) [reply utils/createFilterData.js]: fixed

File: utils/sortProducts.js (hunk 4 above)
Comment (Ksenaaa) [reply utils/sortProducts.js]: fixed

File: utils/createFilterData.js (hunk 2 above)
Comment (Ksenaaa) [reply utils/createFilterData.js]: fixed

File: routes/product.routes.js (hunk 1 above)
Comment (Ksenaaa) [reply routes/product.routes.js]: fixed

**Review by vell28 — APPROVED** [review]

**Pull request merged**